uiowa/uiowa
# Articles by author view block

Resolves #4261 

# How to test

- Code review

```
ddev blt ds --site=91stmeridian.iwp.uiowa.edu && ddev drush @iwp91stmeridian.local uli admin/config/sitenow/sitenow-people
```
- Turn on articles authored by listing on people config page. 
- Go to https://iwp91stmeridian.uiowa.ddev.site/people/chris-keulemans
- Teaser should only show title, date, and issue - and link directly if that is checked. 
  - Test all variations to make sure nothing else shows in the article teaser like source link, teaser, image, etc. 
  - Block should not show on person nodes without related authorship. 

![Screenshot 2023-06-28 at 11 26 17 AM](https://github.com/uiowa/uiowa/assets/1036433/6a642133-a9c6-4a01-bd78-46940a778c4c)


# How to test the splits and features

- [x] Event feature (test on ISWF below)
- [x] isa.uiowa.edu
- [x] now.uiowa.edu
- [x] uipress.uiowa.edu
- [x] admissions.uiowa.edu
- [x] immuno.grad.uiowa.edu
- [x] grad_person_extended (test on default site below)
- [x] sppa.uiowa.edu  (only using `config_split.patch`, no testing instructions below but testing instructions are below)
- [x] pharmacy.uiowa.edu (only using `config_split.patch`, no testing instructions below but testing instructions are below)


## grad_person_extended
```
ddev blt ds --site=default && ddev drush @default.local config-split:activate grad_person_extended  -y && ddev drush @default.local cim && ddev drush @default.local uli admin/config/sitenow/uiowa-articles
```

1. Enable `grad_person_extended`: /admin/config/development/configuration/config-split
2. Import config `ddev drush @default.local cim`

1. Enable articles authored by listing on people config page checkbox
2. Add an author to an article
3. Check that author's person page for the article list


## **immuno.grad.uiowa.edu**
```
ddev blt ds --site=immuno.grad.uiowa.edu && ddev drush @gradimmuno.local uli admin/config/sitenow/uiowa-articles
```
1. Enable articles authored by listing on people config page checkbox
2. Add an author to an article
3. Check that author's person page for the article list

##  **UIPress**
```
ddev blt ds --site=uipress.uiowa.edu && ddev drush @uipress.local uli admin/config/sitenow/uiowa-articles
```

1. Enable articles authored by listing on people config page checkbox
2. Add an author to an article
3. Check that author's person page for the article list

## **isa.uiowa.edu**
```
ddev blt ds --site=isa.uiowa.edu && ddev drush @isa.local uli admin/config/sitenow/uiowa-articles
```
1. Enable articles authored by listing on people config page checkbox
2. Add an author to an article
3. Check that author's person page for the article list

## **now.uiowa.edu**
```
ddev blt ds --site=now.uiowa.edu && ddev drush @now.local uli admin/config/sitenow/uiowa-articles
```
1. Enable articles authored by listing on people config page checkbox
2. Add an author to an article
3. Check that author's person page for the article list
4. Edit person node and add point of contact
5. Save the page and verify point of contact is rendering in main content area

## **ISWF**
```
ddev blt ds --site=iowasummerwritingfestival.uiowa.edu && ddev drush @iowasummerwritingfestival.local uli admin/config/sitenow/uiowa-articles
```

1. Enable articles authored by listing on people config page checkbox
2. Add an author to an article
3. Check that author's person page for the article list

## **Admissions**
```
ddev blt ds --site=admissions.uiowa.edu && ddev drush @admissions.local uli admin/config/sitenow/uiowa-articles
```

1. Enable articles authored by listing on people config page checkbox
2. Add an author to an article
3. Check that author's person page for the article list

##  **SPPA**
```
ddev blt ds --site=sppa.uiowa.edu && ddev drush @sppa.local uli admin/config/sitenow/uiowa-articles
```
1. Enable articles authored by listing on people config page checkbox
2. Add an author to an article
3. Check that author's person page for the article list

##  **Pharmacy**
```
ddev blt ds --site=pharmacy.uiowa.edu && ddev drush @pharmacy.local uli admin/config/sitenow/uiowa-articles
```
1. Enable articles authored by listing on people config page checkbox
2. Add an author to an article
3. Check that author's person page for the article list
